### PR TITLE
TST: fix warnings and stop ignoring them

### PR DESCRIPTION
**Pull request by bsipocz**
Some of these warnings were masking over real issues, e.g. missing test dependencies or ignored pytest markers. Not everything is fixed yet, but locally I'm down to <20, and now pushing a draft PR to check whether anything comes back up when running the tests in different setup scenarios.

This works towards https://github.com/astropy/astroquery/issues/2057 fixes https://github.com/astropy/astroquery/issues/2053

### CODE REVIEW

File: astroquery/cadc/tests/test_cadctap.py [hunk 1]
```diff
@@ -177,7 +177,9 @@ def raise_for_status(self):
         class CapabilitiesResponse:
             def __init__(self):
                 caps_file = data_path('tap_caps.xml')
-                self.text = open(caps_file, 'r').read()
+                with open(caps_file, 'r') as infile:
+                    text = infile.read()
+                self.text = text
```
Comment (eerovaher) [inline astroquery/cadc/tests/test_cadctap.py:182]: Shorter equivalent using `pathlib.Path`:
```suggestion
                self.text = Path(caps_file).read_text()
```

Comment (eerovaher) [reply astroquery/cadc/tests/test_cadctap.py:182]: `Path.read_text()` includes the context manager, see the source at https://github.com/python/cpython/blob/4e1303b4c9022557e04e33791f5a07c0b918b3c5/Lib/pathlib.py#L1129-L1136
My suggestion is less verbose, but the end result is the same either way.

Comment (bsipocz) [reply astroquery/cadc/tests/test_cadctap.py:182]: thanks. It was a bit hit and miss as it still seems undocumented, either case it's indeed nicer and cleaner, and may even address the one remaining mysterious open file warning coming from the remote! cadc tests (I suppose some patching gets in there somehow, I don't see it when the online tests are run, but their test collection triggers it when skipping should be done instead).

**Comment by bsipocz** [comment]
@pllim  and/or @ceb8 - would you mind to give this a quick review? As you see from the cross-links there have been quite a few things discovered, I fixed some of them already, worked around a few, and currently produced more remote-tests failures than before this PR.
That being said, I would like to get this in nowish rather than dragging on forever and growing it to a size that's unmanageable. Besides 1) the ignores are way more specific now, which results in a bit more robust test setup 2) the additional remote failures are still note affecting the users more than the current `main`, and we can aim to fix them gradually.

File: astroquery/alfalfa/tests/test_alfalfa.py [hunk 2]
```diff
@@ -38,7 +38,9 @@ def patch_get_readable_fileobj(request):
     @contextmanager
     def get_readable_fileobj_mockreturn(filename, **kwargs):
         file_obj = data_path(DATA_FILES['spectrum'])  # TODO: add images option
-        yield open(file_obj, 'rb')  # read as bytes, assuming FITS
+        # read as bytes, assuming FITS
+        with open(file_obj, 'rb') as inputfile:
+            yield inputfile
```
Comment (eerovaher) [inline astroquery/alfalfa/tests/test_alfalfa.py:43]: `Path` has a `read_bytes()` method: https://github.com/python/cpython/blob/feb4be1553c947ddb2bd8307419451bfdfc0a6a3/Lib/pathlib.py#L1122-L1127

File: astroquery/ipac/ned/tests/test_ned.py [hunk 3]
```diff
@@ -47,12 +48,14 @@ def patch_get(request):
...
 def patch_get_readable_fileobj(request):
+    @contextmanager
     def get_readable_fileobj_mockreturn(filename, cache=True, encoding=None,
                                         show_progress=True):
         # Need to read FITS files with binary encoding: should raise error
         # otherwise
         assert encoding == 'binary'
-        return open(data_path(DATA_FILES['image']), 'rb')
+        with open(data_path(DATA_FILES['image']), 'rb') as infile:
+            yield infile
```
Comment (pllim) [inline astroquery/ipac/ned/tests/test_ned.py:58]: I guess it is okay to have a context manager inside a context manager, though such is not used in the official example usage at https://docs.python.org/3/library/contextlib.html#contextlib.contextmanager . 🤷 

```suggestion
        infile = open(data_path(DATA_FILES['image']), 'rb')
        try:
            yield infile
        finally:
            infile.close()
```

File: astroquery/sdss/core.py [hunk 4]
```diff
@@ -935,7 +937,16 @@ def _parse_result(self, response, verbose=False):
...
 
-        arr = Table.read(response.text, format='ascii.csv', comment="#")
+        with warnings.catch_warnings():
+            # Capturing the warning and converting the objid column to int64 is necessary for consistency as
+            # it was convereted to string on systems with defaul integer int32 due to an overflow.
+            if sys.platform.startswith('win'):
+                warnings.filterwarnings("ignore", category=AstropyWarning,
+                                        message=r'OverflowError converting to IntType in column.*')
+            arr = Table.read(response.text, format='ascii.csv', comment="#")
+            for id_column in ['objid', 'specobjid']:
```
Comment (pllim) [inline astroquery/sdss/core.py:947]: Nitpicky nitpick. 
```suggestion
            for id_column in ('objid', 'specobjid'):
```

**Review by pllim — APPROVED** [review]
Great clean-up! Just minor comments from drive-by review. Feel free to ignore if they are no applicable.

File: astroquery/alfalfa/tests/test_alfalfa.py (hunk 2 above)
Comment (bsipocz) [reply astroquery/alfalfa/tests/test_alfalfa.py:43]: I know, but tags a reactor for another day

File: astroquery/cadc/tests/test_cadctap_remote.py [hunk 5]
```diff
@@ -293,3 +293,20 @@ def test_list_jobs(self):
...
+        output_file = Path(tmp_path, 'my_observations.xml')
+        cadc.exec_sync("SELECT TOP 3 observationID FROM caom2.Observation",
+                       output_file=output_file)#'my_observations.xml')
+#        assert output_file.exists()
+
+        # now use them to join with the remote table
+        results = cadc.exec_sync(
+            "SELECT o.observationID, intent FROM caom2.Observation o JOIN "
+            "tap_upload.test_upload tu ON o.observationID=tu.observationID",
+            uploads={'test_upload': output_file})#'my_observations.xml'})
```
[the remark this answers is not in the document]
Comment (bsipocz) [reply astroquery/cadc/tests/test_cadctap_remote.py:310]: Thanks for spotting, these are debug remains indeed

File: astroquery/cadc/tests/test_cadctap.py [hunk 6]
```diff
@@ -406,7 +407,7 @@ def test_get_images():
        Mock(side_effect=lambda x, y, z: ['https://some.url']))
 def test_get_images_async():
     with patch('astroquery.utils.commons.get_readable_fileobj', autospec=True) as readable_fobj_mock:
-        readable_fobj_mock.return_value = open(data_path('query_images.fits'), 'rb')
+        readable_fobj_mock.return_value = Path(data_path('query_images.fits'))
```
[the remark this answers is not in the document]
Comment (bsipocz) [reply astroquery/cadc/tests/test_cadctap.py:410]: Apparently it is. And this mocking really bothers me, this change was necessary as a test in the remote file was failing without this, so there is some fundamental problems with the approach. I might have opened an issue for it and we'll try to remember to check when I get in the office.

File: astroquery/cadc/tests/test_cadctap_remote.py [hunk 7]
```diff
@@ -293,3 +293,20 @@ def test_list_jobs(self):
         if len(jobs) > 5:
             jobs_subset = cadc.list_async_jobs(last=5)
             assert len(jobs_subset) == 5
+
+    @pytest.mark.xfail(reason='https://github.com/astropy/astroquery/issues/2538')
```
[the remark this answers is not in the document]
Comment (bsipocz) [reply astroquery/cadc/tests/test_cadctap_remote.py:297]: Eventually, I want these to be cleaned up, and we won't notice if it's a filterwarning.

File: astroquery/ipac/ned/tests/test_ned.py (hunk 3 above)
Comment (bsipocz) [reply astroquery/ipac/ned/tests/test_ned.py:58]: I recall I've seen it somewhere, maybe in the pytest or mock docs or a very similar SO problem. I would rather keep it than switch to try/finally and manual close.

**Review by ceb8 — APPROVED** [review]
LGTM

File: astroquery/cadc/tests/test_cadctap.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (bsipocz) [reply astroquery/cadc/tests/test_cadctap.py:410]: hmm, not exactly interchangeable as it turns out here: https://github.com/astropy/astroquery/pull/2541, but I think it's close enough to be OK. Something is fundamentally wrong with these in-place patches for cadc, but I won't have time to investigate.

[the remark this answers is not in the document]
Comment (ceb8) [reply astroquery/cadc/tests/test_cadctap.py:410]: Worth a bug ticket?

[the remark this answers is not in the document]
Comment (bsipocz) [reply astroquery/cadc/tests/test_cadctap.py:410]: There have been a few genuine bugs smoked out for the module, and I'm not sure it's worth adding the noise, this only affects the testing and the only effect I saw is the appearance of unclosed file warnings in the skipped remote tests. 
I would expect that if there is a real issue with the code, we would be able to catch it with e.g. switching to use ptytest-randomly.

[the remark this answers is not in the document]
Comment (bsipocz) [reply astroquery/cadc/tests/test_cadctap.py:410]: (in short: the only effect I saw is the open file warnings, which is annoying but I doubt would actually affecting users)

**Comment by bsipocz** [comment]
Hmm, so this smoked out a lot of issues on windows, we have 127 remote-test failures there. Some are trivial, but I expect to find a few actual bugs, too.

**Comment by bsipocz** [comment]
Log is here, in case anyone cares to dive in: https://github.com/astropy/astroquery/actions/runs/3171737942/jobs/5165493118#step:5:26169

**Comment by pllim** [comment]
I think it is from astropy-helpers .

**Comment by bsipocz** [comment]
I didn't mean to post a line link, but all the 127 failures, most of which are due to warnings. Updated the link about to point to a better line.

**Comment by pllim** [comment]
So some definitely are bugs because the code failed to account for Windows path using `\\` instead of `/`.

Things like ResourceWarning can probably be ignored for now.

... And this one is really weird: 

```
requests.exceptions.HTTPError: 500 Server Error: Internal Server Error for url:
http://hst.esac.esa.int/ehst-sl-server/servlet/data-action?OBSERVATION_ID=J6FL25S4Q&USERNAME=ehst-astroquery&CALIBRATION_LEVEL=RAW
D:\a\astroquery\astroquery\docs\esa\hubble\hubble.rst:41: UnexpectedException
```

When I copy-pasted http://hst.esac.esa.int/ehst-sl-server/servlet/data-action?OBSERVATION_ID=J6FL25S4Q&USERNAME=ehst-astroquery&CALIBRATION_LEVEL=RAW on my Chrome browser on Windows, a tab popped up but then immediately closed. Might be a server-side problem.

**Comment by pllim** [comment]
And there is also this:

```
TimeoutError: [WinError 10060] A connection attempt failed because the connected party
did not properly respond after a period of time, or established connection failed because
connected host has failed to respond
```

Are some servers purposely blocking requests from Windows? Cert issue?

**Comment by pllim** [comment]
This one might be worth investigating in case it is `io.ascii` bug on Windows:

```
>               result_s = eso.query_surveys(survey, coord1=266.41681662,
                                             coord2=-29.00782497,
                                             box='01 00 00',
                                             cache=False)
...
>                   warnings.warn(
                        "OverflowError converting to {} in column {}, reverting to String."
                        .format(converter_type.__name__, col.name), AstropyWarning)
E                   astropy.utils.exceptions.AstropyWarning: OverflowError converting to IntType in column Object, reverting to String.

...\astropy\io\ascii\core.py:1098: AstropyWarning
```

**Comment by bsipocz** [comment]
A lot of the ESA 500 got fixed upstream yesterday, so those are red herrings.

But the sdss ones e.g. real, and is due to the usual large numbers get casted to str on 32bit. Some of it got worked around to manually convert bach, but apparently I missed a few.

**Comment by pllim** [comment]
Some of the errors might be due to Windows endline vs *NIX endline (since I don't see any path problem in the traceback) but I am not 100% sure.

**Comment by bsipocz** [comment]
> Some of the errors might be due to Windows endline vs *NIX endline

Oh, that's interesting. If that's the case that's an upstream bug for WHITESPACE_NORMALIZE?

**Comment by pllim** [comment]
Re: https://github.com/astropy/astroquery/pull/2520#issuecomment-1267343727

Not sure. Will need someone to investigate. I was thinking more like maybe the query is malformed in the backend... 💭

**Pull request merged**